### PR TITLE
bug where sometimes roles are wiped on save

### PR DESCRIPTION
**Pull request by dollschasingmen**

### CODE REVIEW

**Comment by dollschasingmen** [comment]
@rfroetscher

File: app/models/query.rb [hunk 1]
```diff
@@ -57,7 +57,7 @@ def roles
 
   def set_roles(roles)
     query_roles.delete_all
-    roles.each { |role| query_roles.build(role: role) }
+    roles.each { |role| query_roles.create(role: role) }
```
Comment (rfroetscher) [inline app/models/query.rb]: wonder if you just want to save after all are built?

Comment (dollschasingmen) [inline app/models/query.rb]: hm, that sounds like what we are doing now that is causing the bug?

Comment (rfroetscher) [inline app/models/query.rb]: i mean if you just appended a line that was `save!` after this one

Comment (rfroetscher) [inline app/models/query.rb]: done the current way does it make a db call for each role?

**Comment by rfroetscher** [comment]
yeah, this both solves the bug and is a more truthful way of saving where we only mess with roles after validation :+1:

**Pull request merged**